rust/scannerctl/src/interpret/mod.rs

```diff
@@ -171,6 +171,37 @@ where
     Ok(result)
 }
 
+fn create_loader_from_feed_json(store: &DefaultDispatcher, path: PathBuf) -> Result<FSPluginLoader<PathBuf>, CliError>
+{
+    let result = FSPluginLoader::new(path.clone());
+    let mut file = match File::open(path.as_path()) {
+        Ok(file) => file,
+        Err(e) => {
+            return Err(CliError{
+                filename: path.to_owned().to_string_lossy().to_string(),
+                kind: CliErrorKind::LoadError(LoadError::Dirty(e.to_string()))
+            });
+                    
+        }
+    };
+
+    let mut buf = String::new();
+    if let Err(err) = file.read_to_string(&mut buf) {
+            return Err(CliError{
+                filename: path.to_owned().to_string_lossy().to_string(),
+                kind: CliErrorKind::LoadError(LoadError::Dirty(err.to_string()))
+            });
+        }
```

Comment:
I think this kind of error conversion logic really distracts from the actual thing going on. I initially thought this conversion should just become a `From<io::Error>` impl for `CliError` but I now see that this already exists (and converts to a `Corrupt` variant. I don't know if this is intended, but let's not touch that for now).

I personally would prefer a `load_error` constructor on `CliError` that takes an `io::Error`. If we then also use `fs::read_to_string` which does exactly what `File::open` and `File::read_to_string` do, but in a single function, we end up with

```rust
    let buf = fs::read_to_string(&path).map_err(|e| CliError::load_error(e, &path))?;
    let vts: Vec<storage::item::Nvt> = serde_json::from_str(&buf)?;
```
which I find much much easier to read.

The only thing thats needed is adding the `load_error` method on `CliError`:
```rust
    pub fn load_error(err: std::io::Error, path: &Path) -> Self {
        Self {
            filename: path.to_owned().to_string_lossy().to_string(),
            kind: CliErrorKind::LoadError(LoadError::Dirty(err.to_string())),
        }
    }
```